room/compiler-processing/src/main/java/androidx/room/compiler/processing/InternalXAnnotation.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.room.compiler.processing
+
+import java.lang.annotation.Repeatable
+
+@PublishedApi
+internal interface InternalXAnnotation : XAnnotation {
+    fun <T : Annotation> asAnnotationBox(annotationClass: Class<T>): XAnnotationBox<T>
+}
+
+/**
+ * If this represents a repeatable annotation container this will return the repeated annotations
+ * nested inside it.
+ */
+internal fun XAnnotation.unwrapRepeatedAnnotationsFromContainer(): List<XAnnotation>? {
+    return try {
+        // The contract of a repeatable annotation requires that the container annotation have a single
+        // "default" method that returns an array typed with the repeatable annotation type.
+        val nestedAnnotations = getAsAnnotationList("value")
+
+        // Ideally we would read the value of the Repeatable annotation to get the container class
+        // type and check that it matches "this" type. However, there seems to be a KSP bug where
```

Comment:
can you link the bug here?

Reply:
I think this is the right one https://github.com/google/ksp/issues/358